mkosi/__init__.py

```diff
@@ -725,11 +725,24 @@ def determine_partition_table(args: CommandLineArguments) -> Tuple[str, bool]:
     return table, run_sfdisk
 
 
-def create_image(args: CommandLineArguments, root: str, for_cache: bool) -> Optional[BinaryIO]:
+def exec_sfdisk(args: CommandLineArguments, f: BinaryIO) -> None:
+
+    table, run_sfdisk = determine_partition_table(args)
+
+    if run_sfdisk:
+        run(["sfdisk", "--color=never", f.name], input=table.encode("utf-8"))
+        run(["sync"])
+
```

Comment:
Comparing with `insert_partition` should a call to `partx --update` be added here or in `refresh_partitions` added as well to reread labels from the partition table?

Reply:
So, the "partx --update" is just synchronously synthesizes kernel partition block devices from userspace. But we don't need that here. there partition devices already exist, we just refresh the uuids and labels here, but the block layer doesn't care about that.

The partx call is a bit icky btw, we shouldn't need that, udev asynchronously retiggers a partition rescan anyway. There's no nice way to wait for that though... partx register partitions from userspace, and does so synchronously, which is a bit messy, since it duplicates the partition parser in userspace. The reason it helps here is because it operates synchronously...